quarkus-calculator/solver/src/main/java/com/redhat/training/SolverResource.java

```diff
@@ -0,0 +1,55 @@
+package com.redhat.training;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import com.redhat.training.service.AdderService;
+import com.redhat.training.service.MultiplierService;
+import com.redhat.training.service.SolverService;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SolverResource implements SolverService {
+    final Logger log = LoggerFactory.getLogger(SolverResource.class);
+
+    @Inject
+    @RestClient
+    AdderService adderService;
+
+    @Inject
+    @RestClient
+    MultiplierService multiplierService;
+
+    static final Pattern multiplyPattern = Pattern.compile("(.*)\\*(.*)");
+    static final Pattern addPattern = Pattern.compile("(.*)\\+(.*)");
+
+    @Override
+    @GET
+    @Path("{equation}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Float solve(@PathParam("equation") String equation) {
+        log.info("Solving '{}'", equation);
+        try {
+            return Float.valueOf(equation);
+        } catch (NumberFormatException e) {
+            Matcher addMatcher = addPattern.matcher(equation);
```

Comment:
Same as above, a part of me obsesses over preferring loops over multiple ifs (DRY).

Probably not worth the change.

Reply:
Ditto.